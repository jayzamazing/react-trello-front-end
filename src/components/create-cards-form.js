import React from 'react';
import {reduxForm, Field, focus} from 'redux-form';
import Input from './input';
import {required, nonEmpty, length, isTrimmed} from '../validators';

export function CreateCardsForm(props) {
  return (
    <form className="create-cards-area cards-tile"
      onSubmit={
        props.handleSubmit(values => {
        props.onSubmit(values.cardsTitle, values.cardsText, props.cardslist._id, props.cardslist)
      })}>
        <Field
          component={Input}
          type="text"
          name="cardsTitle"
          validate={[required, nonEmpty, isTrimmed]}
          placeholder="Add a card"
          labelclass="remove"
          inputClass="addcards"
        />
        <button className="create-cards-btn btn btn-success"
          type="submit"
          disabled={props.pristine || props.submitting}>
          Add
        </button>
        <button type="button" className="btn btn-default close-cards-btn"
          aria-label="close button" onClick={() => props.close()}>
            <span className="glyphicon glyphicon-remove"
              aria-hidden="true"></span>
        </button>
      </form>
  );
}
export default reduxForm({
    form: 'createcards',
    onSubmitFail: (errors, dispatch) =>
        dispatch(focus('createcards', Object.keys(errors)[0]))
})(CreateCardsForm);
