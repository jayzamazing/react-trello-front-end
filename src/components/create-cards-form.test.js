'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {CreateCardsForm} from './create-cards-form';

describe('create-cards-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const handleSubmit = jest.fn();
    shallow(<CreateCardsForm handleSubmit={handleSubmit}/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<CreateCardsForm handleSubmit={handleSubmit}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on create button and check dispatches
  it('add cards click', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<CreateCardsForm handleSubmit={handleSubmit}/>);
    wrapper.find('[name="cardsTitle"]').at(0).simulate('change', { target: { value: 'testing' } });
    wrapper.find('.create-cards-btn').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
  //click on create button and check dispatches
  it('close button', () => {
    const handleSubmit = jest.fn();
    const closeModal = jest.fn();
    const wrapper = shallow(<CreateCardsForm close={closeModal} handleSubmit={handleSubmit}/>);
    wrapper.find('[aria-label="close button"]').simulate('click');
    expect(closeModal).toHaveBeenCalled();
  });
});
