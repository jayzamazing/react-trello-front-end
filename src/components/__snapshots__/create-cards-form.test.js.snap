// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-cards-form snapshot 1`] = `
<form
  className="create-cards-area cards-tile"
>
  <Field
    component={[Function]}
    name="cardsTitle"
    placeholder="Add a card"
    textareaClass="addcards"
    validate={
      Array [
        [Function],
        [Function],
        [Function],
      ]
    }
  />
  <button
    className="create-cards-btn btn btn-success"
    type="submit"
  >
    Add
  </button>
  <button
    aria-label="close button"
    className="btn btn-default close-cards-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-remove"
    />
  </button>
</form>
`;
